VCFv4.3.tex

```diff
@@ -503,41 +503,100 @@ \subsection{VCF tag naming conventions}
 
 \section{INFO keys used for structural variants}
 \label{sv-info-keys}
-The following INFO keys are reserved for encoding structural variants. In general, when these keys are used by imprecise variants, the values should be best estimates. When a key reflects a property of a single alt allele (e.g. SVLEN), then when there are multiple alt alleles there will be multiple values for the key corresponding to each alelle (e.g. SVLEN=-100,-110 for a deletion with two distinct alt alleles).
+The following INFO keys are reserved for encoding structural variants. When a key reflects a property of a single alt allele (e.g. SVLEN), then when there are multiple alt alleles there will be multiple values for the key corresponding to each alelle (e.g. SVLEN=-100,-110 for a deletion with two distinct alt alleles).
 \footnotesize
 \begin{verbatim}
 ##INFO=<ID=IMPRECISE,Number=0,Type=Flag,Description="Imprecise structural variation">
 ##INFO=<ID=NOVEL,Number=0,Type=Flag,Description="Indicates a novel structural variation">
 ##INFO=<ID=END,Number=1,Type=Integer,Description="End position of the variant described in this record">
 \end{verbatim}
 \normalsize
-For precise variants, END is POS + length of REF allele - 1, and the for imprecise variants the corresponding best estimate.
+For precise variants, END is POS + length of REF allele - 1.
 
 \footnotesize
 \begin{verbatim}
 ##INFO=<ID=SVTYPE,Number=1,Type=String,Description="Type of structural variant">
 \end{verbatim}
 \normalsize
-This key can be derived from the REF/ALT fields but is useful for filtering. The reserved values must be used for the types listed below:
+SVTYPE is required for all structural variants. The following reserved values should be used whenever possible; indented values are subtypes that should be used if evidence supports them (see examples in Section 5.3, Encoding Structural Variants):\vspace{0.3cm}\\
```

Comment:
"should be used whenever possible" is a massive weakening from "must be used". This weakening makes makes this field useless for downstream analysis programs because there's no requirement that the values actually be one of these. Any value would be spec-compliant.

Practical applications of the restrictive usage of this field include my StructuralVariantAnnotation R package in which I take most popular callers, and (in theory) any VCFv4.3 compliant SV callers and convert them to a standard breakend notation so variant calls written using different notations can actually be compared. With this change, program B will no longer be able to complain that the problem is in the non-compliant output of program A. If two programs don't work with each other then the specs need to be able to be used to say which of the programs is incorrect.

Reply:
Point taken. I will re-write this to clarify. However I still see some ambiguity in the original text:
> This key can be derived from the REF/ALT fields but is useful for filtering. The reserved values must be used for the types listed below:

To take these two sentences in reverse order:

> The reserved values must be used for the types listed below:

One reading of this leaves the door open for SVTYPE to be assigned any value, so long as the author feels their variant is not one of "the types listed below." If the intention is instead, as I believe, to establish a **closed** list of reserved types, such that all VCF SV records require the SVTYPE field, and only the listed values are allowed (and no colons w/ subtypes, etc.) then this sentence needs to change.

> This key can be derived from the REF/ALT fields but is useful for filtering.

This seems to imply the SVTYPE field is redundant with, and secondary to, REF and ALT – and may therefore be optional. Again, I propose this be rewritten or removed.

Reply:
>This seems to imply the SVTYPE field is redundant with, and secondary to, REF and ALT – and may therefore be optional

You raise a very good point. I've been using SVTYPE, but s1.4.5 does indeed define a restrictive set of first-level SV ALT alleles. Although the spec does allow other ALT alleles, it is clear that only ALT alleles in the list are to be considered SVs^.

^ Section 1.4.5 needs the word "imprecise" removed for this to actually be true.

Reply:
It may be convenient for a human to have access for such information (SVTYPE) if the caller decides to output BND records for all variants.
Convenience is a virtual.